docs/tasks/ods-package-image.adoc

```diff
@@ -55,6 +59,11 @@ The following artifacts are generated by the task and placed into `.ods/artifact
 | Reference of the image stream buildah will produce. If not set, the value of `.ods/component` is used.
 
 
+| extra-tags
+| 
+| Additional image tags (e.g. 'latest dev') for pushed images. The primary tag is based on the commit sha. Only tags currently missing from the image will be added.
```

Comment:
I am wondering what the best approach here is. For a tag named sth. like `latest` I think a common expectation would be that it is written for each build. In my eyes, the same applies to tags that reflect the branch name (e.g. `dev` from the example above).

Should we simply always add the tag?

I understand "Docker tags" more like "Git branches", not like "Git tags"  ...

Reply:
Good catch! I overlooked with the lasted revisions the tags would not be applied even if the sha changes, as would happen with a new pipeline. 
On the other hand I wanted to not apply the extra-tags if one reruns a pipeline to avoid timing issues and because it might undo updates from other pipelines.  What is your thoughts on the second concern? Do you have a proposal?

Reply:
We could tag whenever there isn't an artefact file already? The downside of that approach is that when the files are cleaned up, the tag may be re-applied.

Another idea I had is that we could reuse the result from the first check if the GitCommitSha image tag is already built. But that would fail if you have a pipeline that runs just for a certain branch, and the only job of that pipeline is to apply the separate tag ...

I guess both approaches are somewhat flawed, I would lean towards the former as the second blocks an interesting use case I believe.

Reply:
> We could tag whenever there isn't an artefact file already? The downside of that approach is that when the files are cleaned up, the tag may be re-applied.

Are artifacts cleaned up as per retention policies defined by a nexus admin as indicated in this issue #71? Are there 
other cases we should consider?

> Another idea I had is that we could reuse the result from the first check if the GitCommitSha image tag is already built. 
> ... as the second blocks an interesting use case I believe.

Perhaps this could be extended by always applying specified extra-tags if they are what caused the pipeline to be triggered. This is referring to the case mentioned at https://github.com/opendevstack/ods-pipeline/issues/608#issuecomment-1335035973). Would this cover the use case you had in mind? Is it too convoluted?

> ...But that would fail if you have a pipeline that runs just for a certain branch, and the only job of that pipeline is to apply the separate tag ...

Wouldn't this involve a different GitCommitSha so that here the image would be build and the `extra-tags` applied (with the second approach)?